seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json

```diff
@@ -5586,138 +5586,126 @@
       ],
       "type": "waypoint",
       "challengeType": 1,
       "translations": {}
     },
     {
-      "id": "587d7b7e367417b2b2512b22",
-      "title": "Use the parseInt function with a radix",
+      "id": "587d7b7e367417b2b2512b24",
+      "title": "Use the Conditional (Ternary) Operator",
       "description": [
-        "The parseInt() function parses a string and returns an integer. The radix variable defines the number system to convert from.",
-        "The function call looks like:",
-        "parseInt(string, radix);",
-        "var a = parseInt(\"11\", 2);",
-        "Here the radix variable defines that \"11\" is in binary system.",
-        "The above function will convert the string \"11\" to an integer 3.",
-        "Instructions",
-        "Using the parseInt() function, write a function that converts binary numbers to integers and returns them."
+        "The <dfn>conditional operator</dfn>, also called the <dfn>ternary operator</dfn>, is a one line if-else statement.",
+        "The syntax is:",
+        "<code>condition ? statement-if-true : statement-if-false;</code>",
+        "The following function uses an if-else statement to check a condition:",
+        "<blockquote>function findGreater(a, b) {<br>  if(a > b) {<br>    return \"a is greater\";<br>  }<br>  else {<br>    return \"b is greater\";<br>  }<br>}</blockquote>",
+        "This can be re-written using the <code>conditional operator</code>:",
+        "<blockquote>function findGreater(a, b) {<br>  return a > b ? \"a is greater\" : \"b is greater\";<br>}</blockquote>",
+        "<hr>",
+        "Use the <code>conditional operator</code> in the <code>checkEqual</code> function to check if two numbers are equal or not. The function should return either true or false."
       ],
       "challengeSeed": [
-        "function convertToInteger(str) {",
-        "",
+        "function checkEqual(a, b) {",
+        "  ",
         "}",
-        "convertToInteger(\"10011\");"
+        "checkEqual(1, 2);"
       ],
       "tail": [],
       "solutions": [],
       "tests": [
-        "assert(typeof(convertToInteger(\"10011\")) === \"number\", 'message: <code>convertToInteger(\"10011\")</code> should return a number');",
-        "assert(convertToInteger(\"10011\") === 19, 'message: <code>convertToInteger(\"10011\")</code> should return 19');",
-        "assert(convertToInteger(\"111001\") === 57, 'message: <code>convertToInteger(\"111001\")</code> should return 57');",
-        "assert.isNaN(convertToInteger(\"JamesBond\"), 'message: <code>convertToInteger(\"JamesBond\")</code> should return NaN');"
+        "assert(checkEqual(1, 2) === false, 'message: <code>checkEqual(1, 2)</code> should return false');",
+        "assert(checkEqual(1, 1) === true, 'message: <code>checkEqual(1, 1)</code> should return true');",
+        "assert(checkEqual(1, -1) === false, 'message: <code>checkEqual(1, -1)</code> should return false');"
```

Comment:
Possibly add a test to check that the ternary operator was actually used? :smile: